rllib/algorithms/callbacks.py

```diff
@@ -169,7 +171,7 @@ def on_episode_end(
         worker: "RolloutWorker",
         base_env: BaseEnv,
         policies: Dict[PolicyID, Policy],
-        episode: Episode,
+        episode: Union[Episode, EpisodeV2, Exception],
```

Comment:
Can you add to the dosctring why these can also be Exceptions and in which cases?

Reply:
done. oh, yeah, I should have given you a heads up.
while doing this, I noticed that we should probably call on_episode_end() callbacks with the exception if we receive one from the env as well, since otherwise, user logic won't know the episode has errored out, and will left hanging forever.